CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+cmake_minimum_required(VERSION 3.16)
```

Comment:
What's the oldest minimum version we can use here? I am not completely familiar with all the [3.16 changes](https://cmake.org/cmake/help/latest/release/3.16.html) but I'm not seeing any specific features of it that are being used here.

The reason I ask is the older the version of CMake which is supported, the more likely a user will be able to just build the project out of the gate. Less likely to need to setup tools.

Reply:
I'm not totally sure here, but it's probably 3.12 which added object library support for `target_link_libraries`.

There's probably a separate debate to be had about whether this should be a `STATIC` or `OBJECT` library.